### PR TITLE
splitting registry into a separate package

### PR DESCRIPTION
**Pull request by rusenask**
Changed:
* registry is now an interface and doesn't expose its internals
* dotmesh-server/registry.go moved to pkg/registry/registry.go
* types moved to a package that can export them
* finished refactoring user management related RPCs, now using user.UserManager
* added firsts tests & plumbing for the registry package

### CODE REVIEW

File: cmd/dotmesh-server/controller.go [hunk 1]
```diff
@@ -40,7 +41,7 @@ type InMemoryState struct {
 	etcdWaitTimestampLock      *sync.Mutex
 	localReceiveProgress       *Observer
 	newSnapsOnMaster           *Observer
-	registry                   *Registry
+	registry                   registry.Registry
```
Comment (Godley) [inline cmd/dotmesh-server/controller.go]: shouldn't this still be a pointer?

Comment (Godley) [reply cmd/dotmesh-server/controller.go]: Ah, scrolled through and it's changed to an interface, nevermind

File: pkg/registry/registry_test.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
+package registry
```
Comment (Godley) [inline pkg/registry/registry_test.go]: are these tests ran in ci yet?

Comment (rusenask) [reply pkg/registry/registry_test.go]: yes, all tests in the server are run by ci

**Pull request merged**